### PR TITLE
[BE] 관리자 권한 확인 API 구현

### LINKED ISSUE #256: [BE] 관리자 인증 API 구현
## 📄 설명

사용자가 [관리 페이지]에 접근한 경우에 토큰으로 관리자인지 인증합니다. 

**URL**

`POST /api/events/{eventId}/auth`

**요청**

```text
# Cookie
eventToken=jwtToken;
```

**응답**
- 인증된 사용자면 200 OK
- 아니면 401 UNAUTHORIZED

## 🏁 할 일

- [ ] 관리자 인증 API 구현

### PR DESCRIPTION
**Pull request by 3Juhwan**
## issue
- close #256 

## 구현 사항

<h3 id="_행사_어드민_권한_확인 API" style="box-sizing: border-box; margin: 1em 0px 0.5em; padding: 0px; font-family: &quot;Open Sans&quot;, &quot;DejaVu Sans&quot;, sans-serif; font-weight: 300; font-style: normal; color: rgb(204, 204, 204); text-rendering: optimizelegibility; line-height: 1.2; font-size: 1.375em; word-spacing: -0.05em; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">행사 어드민 권한 확인</h3><div class="sect3" style="box-sizing: border-box; margin: 0px; padding: 0px; color: rgb(187, 187, 187); font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h4 id="_행사_어드민_권한_확인_http_request" style="box-sizing: border-box; margin: 1em 0px 0.5em; padding: 0px; font-family: &quot;Open Sans&quot;, &quot;DejaVu Sans&quot;, sans-serif; font-weight: 300; font-style: normal; color: rgb(204, 204, 204); text-rendering: optimizelegibility; line-height: 1.2; font-size: 1.125em; word-spacing: -0.05em;">HTTP request</h4><div class="listingblock" style="box-sizing: border-box; margin: 0px 0px 1.25em; padding: 0px;"><div class="content" style="box-sizing: border-box; margin: 0px; padding: 0px; position: relative;"><pre class="CodeRay highlight nowrap" style="box-sizing: border-box; font-family: &quot;Droid Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: 0.8125em; white-space: pre; margin: 0px; padding: 1em; color: rgba(0, 0, 0, 0.9); line-height: 1.45; text-rendering: optimizespeed; overflow-wrap: normal; background: rgb(247, 247, 248); filter: invert(90%) hue-rotate(180deg); border-radius: 4px; overflow-x: auto;"><code data-lang="http" style="box-sizing: border-box; font-family: &quot;Droid Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: inherit; font-weight: 400; color: inherit; filter: invert(90%) hue-rotate(180deg); line-height: inherit; display: block;">GET /api/events/TOKEN/auth HTTP/1.1
Host: localhost:8080
Cookie: eventToken=토큰토큰
Content-Type: application/x-www-form-urlencoded</code></pre></div></div></div><div class="sect3" style="box-sizing: border-box; margin: 0px; padding: 0px; color: rgb(187, 187, 187); font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h4 id="_행사_어드민_권한_확인_http_response" style="box-sizing: border-box; margin: 1em 0px 0.5em; padding: 0px; font-family: &quot;Open Sans&quot;, &quot;DejaVu Sans&quot;, sans-serif; font-weight: 300; font-style: normal; color: rgb(204, 204, 204); text-rendering: optimizelegibility; line-height: 1.2; font-size: 1.125em; word-spacing: -0.05em;">HTTP response</h4><div class="listingblock" style="box-sizing: border-box; margin: 0px 0px 1.25em; padding: 0px;"><div class="content" style="box-sizing: border-box; margin: 0px; padding: 0px; position: relative;"><pre class="CodeRay highlight nowrap" style="box-sizing: border-box; font-family: &quot;Droid Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: 0.8125em; white-space: pre; margin: 0px; padding: 1em; color: rgba(0, 0, 0, 0.9); line-height: 1.45; text-rendering: optimizespeed; overflow-wrap: normal; background: rgb(247, 247, 248); filter: invert(90%) hue-rotate(180deg); border-radius: 4px; overflow-x: auto;"><code data-lang="http" style="box-sizing: border-box; font-family: &quot;Droid Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: inherit; font-weight: 400; color: inherit; filter: invert(90%) hue-rotate(180deg); line-height: inherit; display: block;">HTTP/1.1 200 OK</code></pre></div></div></div><div class="sect3" style="box-sizing: border-box; margin: 0px; padding: 0px; color: rgb(187, 187, 187); font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h4 id="_행사_어드민_권한_확인_request_cookies" style="box-sizing: border-box; margin: 1em 0px 0.5em; padding: 0px; font-family: &quot;Open Sans&quot;, &quot;DejaVu Sans&quot;, sans-serif; font-weight: 300; font-style: normal; color: rgb(204, 204, 204); text-rendering: optimizelegibility; line-height: 1.2; font-size: 1.125em; word-spacing: -0.05em;">Request cookies</h4>

| Name       | Description       |
|------------|-------------------|
| eventToken | 행사 관리자 토큰 |

</div><div class="sect3 has-source-line data-line-stdin-170" style="box-sizing: border-box; margin: 0px; padding: 0px; color: rgb(187, 187, 187); font-family: &quot;Noto Serif&quot;, &quot;DejaVu Serif&quot;, serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(69, 73, 74); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h4 id="_exceptions_8" style="box-sizing: border-box; margin: 1em 0px 0.5em; padding: 0px; font-family: &quot;Open Sans&quot;, &quot;DejaVu Sans&quot;, sans-serif; font-weight: 300; font-style: normal; color: rgb(204, 204, 204); text-rendering: optimizelegibility; line-height: 1.2; font-size: 1.125em; word-spacing: -0.05em;"><span class="red" style="box-sizing: border-box; color: rgb(191, 0, 0);">Exceptions</span></h4><div class="listingblock has-source-line data-line-stdin-173" style="box-sizing: border-box; margin: 0px 0px 1.25em; padding: 0px;"><div class="content" style="box-sizing: border-box; margin: 0px; padding: 0px; position: relative;"><pre class="CodeRay highlight nowrap" style="box-sizing: border-box; font-family: &quot;Droid Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: 0.8125em; white-space: pre; margin: 0px; padding: 1em; color: rgba(0, 0, 0, 0.9); line-height: 1.45; text-rendering: optimizespeed; overflow-wrap: normal; background: rgb(247, 247, 248); filter: invert(90%) hue-rotate(180deg); border-radius: 4px; overflow-x: auto;"><code data-lang="json" style="box-sizing: border-box; font-family: &quot;Droid Sans Mono&quot;, &quot;DejaVu Sans Mono&quot;, monospace; font-size: inherit; font-weight: 400; color: inherit; filter: invert(90%) hue-rotate(180deg); line-height: inherit; display: block;">[
   {
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">code</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">EVENT_NOT_FOUND</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>,
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">message</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">존재하지 않는 행사입니다.</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>
   },
   {
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">code</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">TOKEN_NOT_FOUND</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>,
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">message</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">토큰이 존재하지 않습니다.</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>
   },
   {
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">code</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">TOKEN_EXPIRED</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>,
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">message</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">만료된 토큰입니다.</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>
   },
   {
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">code</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">TOKEN_INVALID</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>,
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">message</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">유효하지 않은 토큰입니다.</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>
   },
   {
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">code</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">FORBIDDEN</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>,
      <span style="box-sizing: border-box; color: rgb(102, 0, 102);"><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span><span style="box-sizing: border-box;">message</span><span style="box-sizing: border-box; color: rgb(68, 0, 68);">"</span></span>: <span style="box-sizing: border-box; background-color: rgba(255, 0, 0, 0.05);"><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span><span style="box-sizing: border-box; color: rgb(221, 34, 0);">접근할 수 없는 행사입니다.</span><span style="box-sizing: border-box; color: rgb(119, 17, 0);">"</span></span>
   }
]</code></pre></div></div></div>

## 중점적으로 리뷰받고 싶은 부분(선택)
- 인터셉터에서 필요한 모든 검증을 하고 있습니다. 따라서, 컨트롤러에 별도의 로직이 없습니다. 
- 위 내용이 맞는지 검증 부탁드립니다.

### CODE REVIEW

**Review by khabh — APPROVED** [review]
참 잘했어요 👏

File: server/src/main/java/server/haengdong/presentation/EventController.java [hunk 1]
```diff
@@ -90,6 +90,11 @@ public ResponseEntity<Void> loginEvent(
                 .build();
     }
 
+    @PostMapping("/api/events/{eventId}/auth")
```
Comment (kunsanglee) [inline server/src/main/java/server/haengdong/presentation/EventController.java:93]:
```suggestion
    @GetMapping("/api/events/{eventId}/auth")
```
get이 적절한 것 같아요

File: server/src/test/java/server/haengdong/docs/EventControllerDocsTest.java [hunk 2]
```diff
@@ -277,4 +277,27 @@ void findActions() throws Exception {
                         )
                 );
     }
+
+    @DisplayName("행사 어드민 권한을 확인한다.")
+    @Test
+    void authenticateTest() throws Exception {
+        String token = "TOKEN";
+        mockMvc.perform(post("/api/events/{eventId}/auth", token)
```
Comment (kunsanglee) [inline server/src/test/java/server/haengdong/docs/EventControllerDocsTest.java:285]:
```suggestion
        mockMvc.perform(get("/api/events/{eventId}/auth", token)
```

**Review by kunsanglee — APPROVED** [review]
고생하셨습니다 👍

**Pull request merged**